File: Timofeev-PE-42/Bizzare Underground C++/Project1/Character.cpp
#include "Character.h"
#include <iostream>

Character::Character() : isAlive(true), speedCharacter(5), isHanged(false) {
    // ”станавливаем позицию, использу€ сеттер
    setPositionX(10);
    setPositionY(10);
    // ”станавливаем спрайт, использу€ сеттер
    setSprite("player_sprite.png");
    std::cout << "--- Character Initialized ---" << std::endl;
    std::cout << "  Position: (" << getPositionX() << ", " << getPositionY() << ")" << std::endl;
    std::cout << "  Sprite: '" << getSprite() << "'" << std::endl;
    std::cout << "  Is Alive: " << std::boolalpha << getIsAlive() << std::endl;
    std::cout << "  Speed: " << getSpeedCharacter() << std::endl;
    std::cout << "  Is Hanged: " << std::boolalpha << getIsHanged() << std::endl;
}

Character::~Character() {
    std::cout << "--- Character Destroyed ---" << std::endl;
    std::cout << "  Final Position: (" << getPositionX() << ", " << getPositionY() << ")" << std::endl;
    std::cout << "  Final State: Alive = " << std::boolalpha << getIsAlive() << std::endl;
}

void Character::movementRight() {
    std::cout << "Character::movementRight() called. Current X: " << getPositionX() << std::endl;
    setPositionX(getPositionX() + getSpeedCharacter());
    std::cout << "Character::movementRight(): Moved Right. New position X: " << getPositionX() << std::endl;
}

void Character::movementLeft() {
    std::cout << "Character::movementLeft() called. Current X: " << getPositionX() << std::endl;
    setPositionX(getPositionX() - getSpeedCharacter());
    std::cout << "Character::movementLeft(): Moved Left. New position X: " << getPositionX() << std::endl;
}

void Character::jump() {
    std::cout << "Character::jump() called. Performing jump." << std::endl;
}

void Character::fallingCharacter() {
    std::cout << "Character::fallingCharacter() called. Current Y: " << getPositionY() << std::endl;
    setPositionY(getPositionY() - 1); // »митаци€ падени€ на 1 единицу
    std::cout << "Character::fallingCharacter(): Falling. New position Y: " << getPositionY() << std::endl;
}

// Getters
bool Character::getIsAlive() const { return isAlive; }
int Character::getSpeedCharacter() const { return speedCharacter; }
bool Character::getIsHanged() const { return isHanged; }

// Setters
void Character::setIsAlive(bool alive) {
    isAlive = alive;
    std::cout << "Character::setIsAlive() called. New state: isAlive = " << std::boolalpha << isAlive << std::endl;
}
void Character::setSpeedCharacter(int speed) {
    speedCharacter = speed;
    std::cout << "Character::setSpeedCharacter() called. New speed: " << speedCharacter << std::endl;
}
void Character::setIsHanged(bool hanged) {
    isHanged = hanged;
    std::cout << "Character::setIsHanged() called. New state: isHanged = " << std::boolalpha << isHanged << std::endl;
}

File: Timofeev-PE-42/Bizzare-Underground-Java/Character.java
public class Character extends GameObject {
    private boolean isAlive;
    private int speedCharacter;
    private boolean isHanged;

    public Character() {
        super(); // Вызываем конструктор базового класса
        this.isAlive = true;
        this.speedCharacter = 5;
        this.isHanged = false;

        // Устанавливаем позицию и спрайт через сеттеры
        setPositionX(10);
        setPositionY(10);
        setSprite("player_sprite.png");

        System.out.println("--- Character Initialized ---");
        System.out.println("  Position: (" + getPositionX() + ", " + getPositionY() + ")");
        System.out.println("  Sprite: '" + getSprite() + "'");
        System.out.println("  Is Alive: " + isAlive);
        System.out.println("  Speed: " + speedCharacter);
        System.out.println("  Is Hanged: " + isHanged);
    }

    @Override
    protected void finalize() throws Throwable {
        System.out.println("--- Character Destroyed ---");
        System.out.println("  Final Position: (" + getPositionX() + ", " + getPositionY() + ")");
        System.out.println("  Final State: Alive = " + isAlive);
        super.finalize();
    }

    public void movementRight() {
        System.out.println("Character.movementRight() called. Current X: " + getPositionX());
        setPositionX(getPositionX() + getSpeedCharacter());
        System.out.println("Character.movementRight(): Moved Right. New position X: " + getPositionX());
    }

    public void movementLeft() {
        System.out.println("Character.movementLeft() called. Current X: " + getPositionX());
        setPositionX(getPositionX() - getSpeedCharacter());
        System.out.println("Character.movementLeft(): Moved Left. New position X: " + getPositionX());
    }

    public void jump() {
        System.out.println("Character.jump() called. Performing jump.");
        // Логика прыжка
    }

    public void fallingCharacter() {
        System.out.println("Character.fallingCharacter() called. Current Y: " + getPositionY());
        setPositionY(getPositionY() - 1); // Имитация падения
        System.out.println("Character.fallingCharacter(): Falling. New position Y: " + getPositionY());
    }

    // Getters
    public boolean getIsAlive() { return isAlive; }
    public int getSpeedCharacter() { return speedCharacter; }
    public boolean getIsHanged() { return isHanged; }

    // Setters
    public void setIsAlive(boolean alive) {
        this.isAlive = alive;
        System.out.println("Character.setIsAlive() called. New state: isAlive = " + isAlive);
    }
    public void setSpeedCharacter(int speed) {
        this.speedCharacter = speed;
        System.out.println("Character.setSpeedCharacter() called. New speed: " + speedCharacter);
    }
    public void setIsHanged(boolean hanged) {
        this.isHanged = hanged;
        System.out.println("Character.setIsHanged() called. New state: isHanged = " + isHanged);
    }
}